api/v1alpha1/validation/securitypolicy_validate.go

```diff
@@ -57,9 +57,28 @@ func validateSecurityPolicySpec(spec *egv1a1.SecurityPolicySpec) error {
 		errs = append(errs, err)
 	}
 
+	if err := ValidateAPIKeyAuth(spec.APIKeyAuth); err != nil {
```

Comment:
There's a bug that `ValidateSecurityPolicy` isn't called at all currently.
I sent another PR to patch that, https://github.com/envoyproxy/gateway/pull/4987.